packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -32,19 +32,19 @@ export type SearchAction = AsyncThunkAction<
   {analyticsType: AnalyticsType.Search},
   void | {},
   AsyncThunkAnalyticsOptions<StateNeededByAnalyticsProvider>
->;
+> & {name: string};
 
 export type CustomAction = AsyncThunkAction<
   {analyticsType: AnalyticsType.Custom},
   {},
   {}
->;
+> & {name: string};
 
 export type ClickAction = AsyncThunkAction<
   {analyticsType: AnalyticsType.Click},
   {},
   {}
->;
+> & {name: string};
```

Comment:
My main gripe with this is we need to add it to every async action. What would happen if someone forgot to add it? and is there any way we can avoid needing to add it?